app/src/js/components/Executions/execution-status-graph.js

```diff
@@ -16,93 +16,72 @@ import { window } from '../../utils/browser';
 // dagre-d3 expects d3 to be attached to the window
 if (process.env.NODE_ENV !== 'test') window.d3 = d3;
 
-class ExecutionStatusGraph extends React.Component {
-  constructor () {
-    super();
-    this.handleClick = this.handleClick.bind(this);
-    this.onShow = this.onShow.bind(this);
-    this.onHide = this.onHide.bind(this);
-    this.state = {
-      showModal: false
-    };
-  }
-
-  componentDidMount () {
-    const {
-      executionStatus: {
-        executionHistory,
-        stateMachine
-      }
-    } = this.props;
+const ExecutionStatusGraph = ({ executionStatus }) => {
+  const { executionHistory, stateMachine } = executionStatus;
+  const g = useRef(null);
+  const [showModal, setShowModal] = useState(false);
 
+  useEffect(() => {
     const workflow = JSON.parse(stateMachine.definition);
     const events = getExecutionEvents(executionHistory);
     const graph = workflowToGraph(workflow);
     addEventsToGraph(events, graph);
-    this.g = draw(graph);
-    this.renderGraph('.visual', this.g);
-  }
```

Comment:
The biggest issue was that the graph rendering was being done in a `componentDidMount` block. That means that whatever `executionStatus`'s value was when it was first passed is what the final graph will display, even if it changes due a new fetch call.

By making this a functional component and moving that logic into a `useEffect`, it will create a graph using the latest version of `executionStatus`